core/app/models/workarea/pricing/calculators/tax_calculator.rb

```diff
@@ -3,8 +3,10 @@ module Pricing
     module Calculators
       class TaxCalculator
         include Calculator
+        delegate :tax_exempt?, to: :user, allow_nil: true
 
         def adjust
+          adjust_tax_exempt_items_tax && return if tax_exempt?
```

Comment:
I'm wondering about plugin effort, like will the tax 3rd party plugins respect this setting if the logic is in this class?

Reply:
Usually the plugin is responsible for handling tax exemption. 
For example: https://github.com/workarea-commerce/workarea-avatax/blob/master/app/models/workarea/user.decorator

Reply:
I'm asking because when this originally came up the question was around whether tax plugins supported this. So are we saying we don't want those plugins to respect this setting and go off their own logic?

Reply:
The tax plugins use a an api call to their respective services to return a list of taxable statuses. This list is derived from the  configuration of the 3rd party.

I think we would need to first decorate the B2B plugin to delegate tax_exempt? to the user model. After that we would need decorate the tax plugins user models tax_exempt? method to respect whatever logic they use to indicate if a user is non taxable or not.